test/fail_compilation/b20011.d

```diff
@@ -0,0 +1,37 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/b20011.d(23): Error: `S1(cast(ubyte)0u).member` is not an lvalue and cannot be modified
+fail_compilation/b20011.d(26): Error: `S2(null).member` is not an lvalue and cannot be modified
+fail_compilation/b20011.d(27): Error: `S2(null).member` is not an lvalue and cannot be modified
+fail_compilation/b20011.d(31): Error: `U1(cast(ubyte)0u, ).m2` is not an lvalue and cannot be modified
+fail_compilation/b20011.d(35): Error: function `b20011.main.assignableByRef(ref ubyte p)` is not callable using argument types `(ubyte)`
+fail_compilation/b20011.d(35):        cannot pass rvalue argument `S1(cast(ubyte)0u).member` of type `ubyte` to parameter `ref ubyte p`
+fail_compilation/b20011.d(36): Error: function `b20011.main.assignableByOut(out ubyte p)` is not callable using argument types `(ubyte)`
```

Comment:
The messages in my PR are better. Yours don't mention anything about constness and are rather confusing

Reply:
I totally disagree.

Reply:
The test cases are in now. Free to you to rework but I still think that it's sub-accurate to just output that the func cant be called. The user would have to think more about the why.